src/main.rs

```diff
@@ -150,13 +162,26 @@ fn main() -> Result<()> {
                 LAST.with(|last| *last.borrow_mut() = now);
                 let datetime: DateTime<Utc> = now.into();
                 let time = datetime.format("%H%M%S.00");
-                let gpgga = format!(
-                    "$GPGGA,{},{:02}{:010.7},{},{:03}{:010.7},{},4,12,1.3,{:.2},M,0.0,M,1.7,0078",
-                    time, lat_deg, lat_min, lat_dir, lon_deg, lon_min, lon_dir, heightf
-                );
-                let checksum = checksum(gpgga.as_bytes());
-                let gpgga = format!("{}*{:X}\r\n", gpgga, checksum);
-                buf.write_all(gpgga.as_bytes()).unwrap();
+                let message = match &opt.area_id {
+                    Some(area_id) => {
+                        let request_counter = match &opt.request_counter {
```

Comment:
You should be able to use `.unwrap_or_default` here instead of the match

Reply:
The match here allows me to coerce a u8 into an empty string if need be. A default() would cause this value to default to 0, populating the field with a value.

This is undesirable for reasons I will send you on Slack

Reply:
Resolved offline. Keeping `match`